kernel/kernel-api/src/test/scala/io/delta/kernel/internal/icebergcompat/IcebergWriterCompatV3MetadataValidatorAndUpdaterSuite.scala

```diff
@@ -0,0 +1,484 @@
+      }
+      assert(e.getMessage.contains(expectedErrorMessageContains))
+    }
+  }
+
+  private def testIncompatibleActiveLegacyFeature(
+      activeFeatureMetadata: Metadata,
+      tableFeature: String): Unit = {
+    Seq(true, false).foreach { isNewTable =>
+      test(s"cannot enable with $tableFeature active, isNewTable = $isNewTable") {
+        val e = intercept[KernelException] {
+          validateAndUpdateIcebergWriterCompatV3Metadata(
+            isNewTable,
+            activeFeatureMetadata,
+            getCompatEnabledProtocol())
+        }
+        assert(e.getMessage.contains(
+          s"Table features [$tableFeature] are incompatible with icebergWriterCompatV3"))
+      }
+    }
+  }
+
+  /* --- CM_ID_MODE_ENABLED and PHYSICAL_NAMES_MATCH_FIELD_IDS_CHECK tests --- */
+
+  Seq(true, false).foreach { isNewTable =>
+    Seq(true, false).foreach { icebergCompatV3Enabled =>
+      test(s"column mapping mode `id` is auto enabled when icebergWriterCompatV3 is enabled, " +
+        s"isNewTable = $isNewTable, icebergCompatV3Enabled = $icebergCompatV3Enabled") {
+
+        val tblProperties = icebergWriterCompatV3EnabledProps ++
+          (if (icebergCompatV3Enabled) {
+             icebergCompatV3EnabledProps
+           } else {
+             Map()
+           })
+        val metadata = testMetadata(cmTestSchema(), tblProps = tblProperties)
+        val protocol = getCompatEnabledProtocol()
+
+        assert(!metadata.getConfiguration.containsKey("delta.columnMapping.mode"))
+
+        if (isNewTable) {
+          val updatedMetadata =
+            validateAndUpdateIcebergWriterCompatV3Metadata(isNewTable, metadata, protocol)
+          assert(updatedMetadata.isPresent)
+          assert(updatedMetadata.get().getConfiguration.get("delta.columnMapping.mode") == "id")
+          // We correctly populate the column mapping metadata
+          verifyCMTestSchemaHasValidColumnMappingInfo(
+            updatedMetadata.get(),
+            isNewTable,
+            true,
+            true)
+        } else {
+          val e = intercept[KernelException] {
+            validateAndUpdateIcebergWriterCompatV3Metadata(isNewTable, metadata, protocol)
+          }
+          assert(e.getMessage.contains(
+            "The value 'none' for the property 'delta.columnMapping.mode' is" +
+              " not compatible with icebergWriterCompatV3 requirements"))
+        }
+      }
+    }
+  }
+
+  test("checks are not enforced when table property is not enabled") {
+    // Violate check by including BYTE type column
+    val schema = new StructType().add("col", ByteType.BYTE)
+    val metadata = testMetadata(schema)
+    assert(!TableConfig.ICEBERG_WRITER_COMPAT_V3_ENABLED.fromMetadata(metadata))
+    validateAndUpdateIcebergWriterCompatV3Metadata(
+      true, /* isNewTable */
+      metadata,
+      getCompatEnabledProtocol())
+  }
+
+  Seq("name", "none").foreach { cmMode =>
```

Comment:
seems like this test is common with v1writer. Can we refactor the suite? Need not be in this PR.